scripts/FigS7_cellassign-justification.R

```diff
@@ -0,0 +1,152 @@
+
+# read in sce objects
+processed_sce <- readr::read_rds(processed_sce_file)
+
+# define output file paths
+# use png for UMAP 
+png_dir <- here::here("figures", "pngs")
+umap_png_file <- file.path(png_dir, "FigS7A_cellassign-umap.png")
+
+# use pdf for heatmap 
+pdf_dir <- here::here("figures", "pdfs")
+heatmap_pdf_file <- file.path(pdf_dir, "FigS7B_cellassign-submitter-heatmap.pdf")
+
+# source in helper functions for plotting
+function_file <- here::here("scripts", "utils", "celltype-plot-helper-functions.R")
+source(function_file)
+
+# UMAP -------------------------------------------------------------------------
+# Create data frame of cell types
+celltype_df <- create_celltype_df(processed_sce)
+
+# lump celltypes for cellassign 
+celltype_df <- celltype_df |> 
+  dplyr::mutate(cellassign_celltype_annotation_lumped = cellassign_celltype_annotation |> 
+                  forcats::fct_lump_n(5, other_level = "All remaining cell types", ties.method = "first") |>
+                  forcats::fct_infreq() |>
+                  forcats::fct_relevel("Unknown cell type", "All remaining cell types", after = Inf)
+  )
+
+# faceted UMAP
+faceted_umap <- ggplot(
+  celltype_df,
+  aes(x = UMAP1, y = UMAP2, color = cellassign_celltype_annotation_lumped)
+) +
+  # set points for all "other" points
+  geom_point(
+    data = dplyr::select(
+      celltype_df, -cellassign_celltype_annotation_lumped
+    ),
+    color = "gray80",
+    alpha = 0.5,
+    size = 0.3
+  ) +
+  # set points for desired cell type
+  geom_point(size = 0.3, alpha = 0.5) +
+  facet_wrap(
+    vars(cellassign_celltype_annotation_lumped),
+    ncol = 3
+  ) +
+  scale_color_brewer(palette = "Dark2") +
+  # remove axis numbers and background grid
+  scale_x_continuous(labels = NULL, breaks = NULL) +
+  scale_y_continuous(labels = NULL, breaks = NULL) +
+  guides(
+    color = guide_legend(
+      title = "Cell types",
+      # more visible points in legend
+      override.aes = list(
+        alpha = 1,
+        size = 2
+      )
+    )
+  ) +
+  theme(legend.position = "bottom")
+
+ggsave(umap_png_file, faceted_umap, width = 8, height = 8)
+
+# Heatmap ----------------------------------------------------------------------
+
+# get jaccard similarity index
+jaccard_submitter_matrix <- make_jaccard_matrix(
+  celltype_df,
+  "submitter_celltype_annotation",
+  "cellassign_celltype_annotation"
+)
+
+# heatmap comparing cellassign to submitter annotations
+heatmap <- ComplexHeatmap::Heatmap(
+  # transpose so submitter will be columns
+  t(jaccard_submitter_matrix),
+  col = circlize::colorRamp2(c(0, 1), colors = c("white", "darkslateblue")),
+  border = TRUE,
+  ## Row parameters
+  cluster_rows = FALSE,
+  row_title = "CellAssign annotations",
+  row_title_gp = grid::gpar(fontsize = 12),
+  row_title_side = "right",
+  row_names_side = "left",
+  row_names_gp = grid::gpar(fontsize = 10),
+  ## Column parameters
+  cluster_columns = FALSE,
```

Comment:
I see that we aren't clustering, but I wonder if we want to look at the sorting of the input matrix for the submitter cell types only to group all of the tumor cell annotations?

Reply:
<img width="957" alt="Screenshot 2024-03-12 at 11 36 38 AM" src="https://github.com/AlexsLemonade/scpca-paper-figures/assets/54039191/2a3c9312-6a9e-4f3d-bf36-20f074741a7a">

This is sorted so that everything with `tumor` is first and then all other cell types. @jashapiro is this what you had in mind?